src/AzureDevOpsPolicyConfigurator/Arguments/Commands/CommandBase.cs

```diff
@@ -33,12 +35,42 @@ public override ValidationResult Validate(CommandContext context, TSettings sett
 
             XmlConfigurator.ConfigureAndWatch(logRepository, new FileInfo(fileInfo));
 
-            SetVerbosityLevel(settings, logRepository);
+            this.SetVerbosityLevel(settings, logRepository);
+
+            this.ValidateSettings(settings);
 
             return base.Validate(context, settings);
         }
 
-        private static void SetVerbosityLevel(TSettings settings, ILoggerRepository logRepository)
+        private void ValidateSettings(TSettings settings)
+        {
+            if (string.IsNullOrEmpty(settings.CollectionUrl))
+            {
+                throw new ArgumentValidationException("Argument \"connectionurl\" is missing!");
+            }
+
+            if (settings.Auth == null)
+            {
+                throw new ArgumentValidationException("Argument \"auth\" is missing!");
+            }
+
+            if ((settings.Auth == AuthMethod.Basic || settings.Auth == AuthMethod.OAuth) && string.IsNullOrEmpty(settings.User))
+            {
+                throw new ArgumentValidationException("Argument \"user\" must be provided if \"auth\" is set to Basic or OAuth!");
+            }
+
+            if ((settings.Auth == AuthMethod.Basic || settings.Auth == AuthMethod.OAuth) && string.IsNullOrEmpty(settings.Password))
```

Comment:
Do the check `(settings.Auth == AuthMethod.Basic || settings.Auth == AuthMethod.OAuth)` once.